lib/IDE/Utils.cpp

```diff
@@ -1252,12 +1252,55 @@ Expr *swift::ide::getBase(ArrayRef<Expr *> ExprStack) {
   return Base;
 }
 
-bool swift::ide::isDynamicCall(Expr *Base, ValueDecl *D) {
-  auto TyD = D->getDeclContext()->getSelfNominalTypeDecl();
-  if (!TyD)
+static bool isDeclSyntacticallyOverridable(ValueDecl *D) {
+  auto *NTD = D->getDeclContext()->getSelfNominalTypeDecl();
+  if (!NTD)
     return false;
 
-  if (isa<StructDecl>(TyD) || isa<EnumDecl>(TyD) || D->isFinal())
+  if (!(isa<ClassDecl>(NTD) || isa<ProtocolDecl>(NTD)) || D->isFinal() ||
+      NTD->isFinal() || NTD->isActor())
+    return false;
+
+  if (auto *Accessor = dyn_cast<AccessorDecl>(D)) {
+    switch (Accessor->getAccessorKind()) {
+    case AccessorKind::Get:
+    case AccessorKind::Set:
+      break;
+    case AccessorKind::WillSet:
+    case AccessorKind::DidSet:
+    case AccessorKind::Address:
+    case AccessorKind::MutableAddress:
+    case AccessorKind::Read:
+    case AccessorKind::Modify:
+      return false;
```

Comment:
Probably doesn’t really matter but don’t we allow overriding these accessors in the same way that we allow overriding get and set?

Reply:
This is just moved from the index code, I was assuming the difference here is that get/set are "called" in actual code. Where as the others are more side effects (ie. didSet runs after setting) 🤷.

Reply:
OK. I don’t have strong opinions either way but if it doesn’t make a difference, we might be able to just remove this `switch` altogether.

Reply:
IIRC I was considering these as just recording in the index data that they are present, as children of a particular property, and that they shouldn't look like "dynamic and overridable" functions.